### PR TITLE
Rootfs packaging division, base, variscite,meticulous

### PR DESCRIPTION
**Pull request by Octavio2121**
The changes I made were to package the rootfs files into three segments: debian-base, Variscite, and Meticulous.
The three tar.gz packages are generated inside the output folder.

### CODE REVIEW

File: build_img.sh [hunk 1]
```diff
@@ -141,6 +141,8 @@ function create_image () {
     dd if=output/imx-boot-sd.bin of=${PARTITION}1 bs=1K status=noxfer status=progress 2>/dev/null
 
     echo "Installing OS                 to ${PARTITION}3"
+    #pv output/rootfs-base.tar.gz | tar -xzp -C sdcard
```
Comment (Mimoja) [inline build_img.sh:144]: If you think is often needed please put a switch to the script or removed the commented out code (I know, I should do the same)

File: var_make_debian.sh [hunk 2]
```diff
@@ -700,6 +701,17 @@ function make_bcm_fw()
 		install -m 0644 ${1}/LICENSE ${2}/lib/firmware/bcm/
 	fi
 	install -m 0644 ${1}/LICENSE ${2}/lib/firmware/brcm/
+
+	echo "Verifying the contents of ${ROOTFS_BASE}:"
+	ls -l ${ROOTFS_BASE}
+	# Repackaging with Variscite customizations.
+	pr_info "Repackaging with Variscite customizations"
+	if [ -d "${ROOTFS_BASE}" ] && [ "$(ls -A ${ROOTFS_BASE})" ]; then
+    tar czvf ${PARAM_OUTPUT_DIR}/rootfs-variscite.tar.gz -C ${ROOTFS_BASE} .
```
Comment (Mimoja) [inline var_make_debian.sh]: nit: indentation

File: variscite/weston_rootfs.sh [hunk 3]
```diff
@@ -28,6 +28,20 @@ function make_debian_weston_rootfs()
 	mount -o bind /sys ${ROOTFS_BASE}/sys
 	chroot $ROOTFS_BASE /debootstrap/debootstrap --second-stage --verbose
 
+	#Package the base Debian image into a tar.gz file
+	pr_info "Packaging the base Debian image"
+	if [ -d "${ROOTFS_BASE}" ] && [ "$(ls -A ${ROOTFS_BASE})" ]; then
+    tar czvf ${PARAM_OUTPUT_DIR}/rootfs-base.tar.gz -C ${ROOTFS_BASE} .
```
Comment (Mimoja) [inline variscite/weston_rootfs.sh]: nit: indentation

File: variscite/weston_rootfs.sh [hunk 4]
```diff
@@ -1030,3 +973,86 @@ EOF
...
+	rm -rf ${ROOTFS_BASE}/usr/lib/python3.12/site-packages/pip*
+	chroot ${ROOTFS_BASE} bash -c "python3.12 -m ensurepip --upgrade --altinstall"
+
+	# Install python requirements for meticulous
+	cp ${G_VARISCITE_PATH}/python-requirements.txt ${ROOTFS_BASE}/tmp/python-requirements.txt
+	chroot ${ROOTFS_BASE} bash -c "python3.12 -m pip install -r /tmp/python-requirements.txt"
+
+	#Installing Meticulous packages
+	#chroot ${ROOTFS_BASE} /bin/bash -c "
+    #for pkg in $(cat meticulous_packages.txt); do
```
Comment (Mimoja) [inline variscite/weston_rootfs.sh]: Please remove commented out code

**Comment by Mimoja** [comment]
If I see it correct: It is currently not possible to only call an individual stage, e.g.
```
./var_make_debian.sh --meticulous
```

This should be added next so that we don't have to run all stages at once but can run them as one

File: build_img.sh [hunk 5]
```diff
@@ -141,6 +141,8 @@ function create_image () {
     dd if=output/imx-boot-sd.bin of=${PARTITION}1 bs=1K status=noxfer status=progress 2>/dev/null
```
Comment (Mimoja) [inline build_img.sh:141]: Commit message:
Please write a small paragraph about what has been changed and why

**Review by Mimoja — CHANGES_REQUESTED** [review]

File: build_img.sh [hunk 6]
```diff
@@ -2,7 +2,7 @@
 set -eo pipefail
 
 declare    IMAGE_TARGET="sdcard.img"
-#declare    IMAGE_TARGET="/dev/sdc"
```
Comment (Mimoja) [inline build_img.sh:5]: We should move that to a commandline flag as well. My bad. Will fix this :D

File: var_make_debian.sh [hunk 7]
```diff
@@ -986,6 +1023,21 @@ case $PARAM_CMD in
...
 		;;
+	debianbase )
+		cmd_make_uboot  &&
+		cmd_make_kernel &&
+		cmd_make_kmodules &&
+		cmd_make_kernel_header_deb &&
+		cmd_make_debian_base_rootfs
+		;;
+	variscite )
+		cmd_make_uboot  &&
```
Comment (Mimoja) [inline var_make_debian.sh:1034]: This will not reuse the old kernel and similiar, but instead generate everything anew

**Review by Mimoja — APPROVED** [review]

File: var_make_debian.sh (hunk 7 above)
Comment (Octavio2121) [reply var_make_debian.sh:1034]: Then, could I add the condition that if the "debian_base" package already exists, it should only unpack and install the Variscite packages, and in the case of wanting to create a complete Meticulous image, a condition that checks if the packages from the previous stages already exist and then only unpack and install the Meticulous packages?

**Pull request merged**